skimage/filters/rank/generic_cy.pyx

```diff
@@ -377,7 +377,7 @@ cdef inline void _kernel_otsu(dtype_t_out* out, Py_ssize_t odepth,
                               Py_ssize_t s0, Py_ssize_t s1) nogil:
     cdef Py_ssize_t i
     cdef Py_ssize_t max_i
-    cdef double P, mu1, mu2, q1, new_q1, sigma_b, max_sigma_b
+    cdef double P, mu1, mu2, q1, new_q1, sigma_b, max_sigma_b, t
```

Comment:
This `t` also needs to be a `double`.

Reply:
Whoops, my mistake.  I think it's only the prior one.

Reply:
Hmm, I thought I fixed those. May have pushed a wrong branch...

Reply:
Indeed, this is not what I was testing locally.